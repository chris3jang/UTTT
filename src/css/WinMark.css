.winMark {
	position: absolute;
	height: 162px;
	width: 162px;
	line-height: 162px;
	font-size: 150px;
	z-index: -2;
	text-align: center;
	top: 0;
	left: 0;
}

.visible {
	opacity: 1;
}

.instant {
	transition: none;
}

.outerTileWinTrans {
	transition-property: opacity;
    transition-duration: 2s;
    transition-delay: 4s;
}

.svgclass {
	height: 100%;
	width: 100%;
}

.bigWinMark {
	position: absolute;
	height: 504px;
	width: 504px;
	line-height: 504px;
	top: 0;
	left: 0;
	right: 0;
	margin: auto;
	z-index: -2;
}

.bigSvgClass {
	height: 100%;
	width: 100%;
}


.hidden {
	opacity: 0;
}

.smallPath {
	stroke-dasharray: 1000;
	stroke-dashoffset: 1000;
	stroke-width: 3;
	animation: smallDash 5s linear forwards;
}

.bigPath {
	stroke-dasharray: 3000;
	stroke-dashoffset: 3000;
	stroke-width: 3;
	animation: bigDash 15s linear forwards;
}

@keyframes smallDash {
	from {
		stroke-dashoffset: 1000;
	}
	to {
		stroke-dashoffset: 0;
	}
}

@keyframes bigDash {
	from {
		stroke-dashoffset: 3000;
	}
	to {
		stroke-dashoffset: 0;
	}
}