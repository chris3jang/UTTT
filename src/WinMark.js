import React, { Component } from 'react';
import './WinMark.css';

const WinMark = ({isBoardSet, boardWinner, finalBoardWinPositions, winID, tileHovered, changedClassName}) => {

  const getWinEl = (id) => {
    const winDim = {
      "012": "M10 29 L152 29",
      "345": "M10 81 L152 81",
      "678": "M10 133 L152 133",
      "036": "M29 10 L29 152",
      "147": "M81 10 L81 152",
      "258": "M132 10 L132 152",
      "048": "M10 10 L152 152" ,
      "246": "M10 152 L152 10"
    };
    const cN = (isBoardSet ? "small" : "big") + "Path" + (finalBoardWinPositions.includes(tileHovered[0]) && finalBoardWinPositions.length !== 0 ? " hidden" : "")
    if((boardWinner !== ' ' && isBoardSet) || (!isBoardSet && finalBoardWinPositions.length !== 0)) {
      return <path d={winDim[id]} stroke="#000000" stroke-width="5" className={cN}></path>
    }
    else return
  }
  

  return (
    <div className={isBoardSet ? changedClassName : "bigWinMark"}>
	    <svg className={isBoardSet ? "svgclass" : "bigSvgClass"}>
	    	<g transform={"scale(" + (isBoardSet ? 1 : 3.1) + ")"}>
	    		{getWinEl(winID)}
	    	</g>
	    </svg>
	</div>
  );

}

export default WinMark;