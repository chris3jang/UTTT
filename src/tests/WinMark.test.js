import React from 'react';
import ReactDOM from 'react-dom';
import WinMark from '../WinMark';

import renderer from 'react-test-renderer';

import { mount, shallow } from 'enzyme';

const defProps = {
	isBoardSet: true,
	boardWinner: ' ',
	finalBoardWinPositions: [],
	winID: "",
	tileHovered: [0,0],
	changedClassName: "winMark"
};

describe('default <WinMark /> content accurately represents mark', () => {
	it('renders a snapshot', () => {
  		const tree = renderer.create(
  			<WinMark
  				{...defProps}>
  			</WinMark>).toJSON();
  		expect(tree).toMatchSnapshot();
	});

	it('renders default props', () => {
		const wrapper = shallow(<WinMark {...defProps} />);
	})
});

it('inner board <WinMark /> <path /> is present and absent when board is and isnt won', () => {
  const wrapper = shallow(<WinMark {...defProps} />);
  expect(wrapper.find('path').exists()).toBe(false);
  wrapper.setProps({ boardWinner: '✕'});
  expect(wrapper.find('path').exists()).toBe(true);
});

it('final board <WinMark /> <path /> is present and absent when board is and isnt won', () => {
  const wrapper = shallow(<WinMark {...defProps} />);
  wrapper.setProps({isBoardSet: false})
  expect(wrapper.find('path').exists()).toBe(false);
  wrapper.setProps({ finalBoardWinPositions: [0, 1, 2]});
  expect(wrapper.find('path').exists()).toBe(true);
});

it('<WinMark /> <path /> has proper attributes for position', () => {
  const wrapper = shallow(<WinMark {...defProps} />);
  wrapper.setProps({ boardWinner: '✕', winID: "012"})
  expect(wrapper.find('path').props()).toHaveProperty('d', "M10 29 L152 29");
  //delete these, if one works they will all work
  wrapper.setProps({ boardWinner: '✕', winID: "345"})
  expect(wrapper.find('path').props()).toHaveProperty('d', "M10 81 L152 81");
  wrapper.setProps({ boardWinner: '✕', winID: "678"})
  expect(wrapper.find('path').props()).toHaveProperty('d', "M10 133 L152 133");
  wrapper.setProps({ boardWinner: '✕', winID: "036"})
  expect(wrapper.find('path').props()).toHaveProperty('d', "M29 10 L29 152");
  wrapper.setProps({ boardWinner: '✕', winID: "147"})
  expect(wrapper.find('path').props()).toHaveProperty('d', "M81 10 L81 152");
  wrapper.setProps({ boardWinner: '✕', winID: "258"})
  expect(wrapper.find('path').props()).toHaveProperty('d', "M132 10 L132 152");
  wrapper.setProps({ boardWinner: '✕', winID: "048"})
  expect(wrapper.find('path').props()).toHaveProperty('d', "M10 10 L152 152");
  wrapper.setProps({ boardWinner: '✕', winID: "246"})
  expect(wrapper.find('path').props()).toHaveProperty('d', "M10 152 L152 10");
});

it('<WinMark /> <path /> has proper attributes for position', () => {
  const wrapper = shallow(<WinMark {...defProps} />);
  wrapper.setProps({ isBoardSet: false, winID: "012", finalBoardWinPositions: [0, 1, 2]})
  expect(wrapper.find('path').props()).toHaveProperty('d', "M10 29 L152 29");
});

//test that when boardWinner becomes either x or o instead of ' ', that 1) winID isn't "", 2) changedClassName has hidden and instant (after transition)







