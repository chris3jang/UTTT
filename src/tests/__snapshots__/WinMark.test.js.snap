exports[`default <WinMark /> content accurately represents mark renders a snapshot 1`] = `
<div
  className="winMark">
  <svg
    className="svgclass">
    <g
      transform="scale(1)" />
  </svg>
</div>
`;
